src/reducers/TemperatureSyncReducer.js

```diff
@@ -24,36 +25,70 @@ const initialState = () => ({
   total: 0,
   syncState: null,
   isSyncing: false,
+  modalIsOpen: false,
+  lastTemperatureSync: null,
+  lastSyncError: null,
+  currentSensorName: null,
 });
 
 export const TemperatureSyncReducer = (state = initialState(), action) => {
   const { type } = action;
 
   switch (type) {
+    case REHYDRATE: {
+      const { payload: previousState } = action;
+      const { temperatureSync: previousTemperatureSyncState } = previousState;
+      const { lastTemperatureSync } = previousTemperatureSyncState;
+
+      return { ...initialState(), lastTemperatureSync };
+    }
+
+    case TEMPERATURE_SYNC_ACTIONS.UPDATE_SENSOR_PROGRESS: {
+      const { payload } = action;
+      const { sensor } = payload;
+      const { name: currentSensorName } = sensor;
+
+      return { ...state, currentSensorName };
+    }
+
+    case TEMPERATURE_SYNC_ACTIONS.OPEN_MODAL: {
+      return { ...state, modalIsOpen: true };
+    }
+    case TEMPERATURE_SYNC_ACTIONS.CLOSE_MODAL: {
+      return { ...state, modalIsOpen: false };
+    }
```

Comment:
Just a general thing, we've got lots of modals which we display/hide using a generic key like `modalIsOpen`. Lots of them in separate bits of the redux store, but still might be good to make the keys a bit more explicit, e.g. `temperatureSyncModalIsOpen` etc.

Reply:
feel like `temperatureSync.temperatureSyncModalIsOpen` is just quite redundant, can always rename in parts of the app where it's needed? `const { modalIsOpen: .. } = ..` 

Also since we're here. I was wondering what your thoughts are on the number of cases in the reducer. Do you think it would be nicer to have a `updateState` action, where the new state is passed in the payload, rather than unique action cases which define what the state will become?

Reply:
> feel like temperatureSync.temperatureSyncModalIsOpen is just quite redundant, can always rename in parts of the app where it's needed? const { modalIsOpen: .. } = ..

Yeah, I'm thinking more for the cases where one page etc. has multiple modals. Agree there's a trade-off with explicitness and redundancy here, though.

> Do you think it would be nicer to have a updateState action, where the new state is passed in the payload, rather than unique action cases which define what the state will become?

I don't feel strongly either way. My initial thought would be that there's a good chance both would be similar as you'll probably need lots of smaller functions to handle the cases specified in the payload (or a big do-it-all method). I think you'd possibly end up with similar functions but calling them as helpers rather than explicit actions?